v2/git-initializer/main.go

```diff
@@ -102,10 +102,62 @@ func gitCheckout() error {
 		auth = publicKeys
 	}
 
+	var worktree *git.Worktree
+	if event.Worker.Git.Commit == "" && event.Worker.Git.Ref == "" {
+		worktree, err = simpleCheckout(gitConfig.CloneURL, auth)
+	} else {
+		worktree, err = complexCheckout(
+			gitConfig.CloneURL,
+			event.Worker.Git.Commit,
+			event.Worker.Git.Ref,
+			auth,
+		)
```

Comment:
The more complex scenario where we need to do some legwork to figure out if a ref was a fulls ref like `/refs/heads/foo` or a more "casual" ref like `foo` is accounted for by `complexCheckout()`. The logic that used to be right here in this function just moved down to there.